### DISCUSSION TITLE
Cannot inject ServletConfig in RESTEasy Reactive

### DISCUSSION THREAD

**Discussion by sponiro** [quarkusio/quarkus#34618] [Q&A] [answered]
Hello everyone,

I have a problem with RESTEasy Reactive. It seems, RESTEasy Reactive does not support injection of a ServletConfig with @Context. Let me explain:

In a Quarkus project we use RESTEasy Classic. We also use Swagger from https://github.com/swagger-api/swagger-core. I tried to switch from RESTEasy Classic to Reactive. When I start the project I get an error like this:

[1] Unsatisfied dependency for type jakarta.servlet.ServletConfig and qualifiers [@Default]

I researched and found that Swagger itself defines resources which inject the ServletConfig. See https://github.com/swagger-api/swagger-core/blob/master/modules/swagger-jaxrs2/src/main/java/io/swagger/v3/jaxrs2/integration/resources/OpenApiResource.java

I tried to reconstruct the situation in a more simple way. I did create a new project:

```bash
quarkus create app quarkus-servlet-config -x io.quarkus:quarkus-resteasy-reactive
```

and modified the GreetingResource to inject the ServletConfig:

```Java
@Path("/hello")
public class GreetingResource {

    @Context
    ServletConfig servletConfig;

    @GET
    @Produces(MediaType.TEXT_PLAIN)
    public String hello() {
        return "Hello from RESTEasy Reactive";
    }
}
```
Then I started it with quarkus dev and it failed the same way. I also tried to add the extensions quarkus-undertow and quarkus-resteasy-reactive-servle but without any positive results.

At the moment, I am not sure that I can expect the @Context annotation to work or if this is just not supported. Does anybody know if this should work and if so what I am missing? Or did I found a bug?

Greetings
Robert

[a comment by an automated account is not shown]

**Reply by geoand** [reply, marked as answer]
ServletConfig (and servler in general) is not supported in RESTEasy Reactive.

Are you using the quarkus-openapi extension?
Asking because that works just fine with RESTEasy Reactive

**Reply by sponiro** [reply]
No, we are not using the quarkus-smallrye-openapi extension. We thought of using that but a migration needs to take place for that from the swagger plugin to the extension. But this is probably the way to go.
If servlets in general are not supported then maybe the migration document for RESTEasy (https://quarkus.io/guides/resteasy-reactive-migration) can mention that for future reference?

Thanks for the help!

**Reply by geoand** [reply]
We can certainly use a note. Would you like to contribute that?

**Reply by sponiro** [reply]
I tried myself on a small note about this topic. See [this pull request](https://github.com/quarkusio/quarkus/pull/34625).

**Comment by rajbhupendra588** [comment]
hi @geoand & @sponiro 

What is the alternative now?

I have swagger integration with Quarkus and now migrating to  resteasy reactive and I am also getting below error 

`
Unsatisfied dependency for type jakarta.servlet.ServletConfig and qualifiers [@Default]
        - java member: io.swagger.v3.jaxrs2.integration.resources.AcceptHeaderOpenApiResource#config
        - declared on CLASS bean [types=[io.swagger.v3.jaxrs2.integration.resources.AcceptHeaderOpenApiResource, java.lang.Object], qualifiers=[@Default, @Any], target=io.swagger.v3.jaxrs2.integration.resources.AcceptHeaderOpenApiResource]
        `
        
       I understand with Quarkus resteasy reactive migration, Servlet is not supported

**Reply by geoand** [reply]
You can use our `quarkus-smallrye-openapi` extension to get Swagger, see [this](https://quarkus.io/guides/openapi-swaggerui)
